### PR TITLE
Add authentication functionality

### LINKED ISSUE #10: A3: Users can log in and log out of their account

### PR DESCRIPTION
**Pull request by Gabau**
Minor bugs: Forgot to include the following earlier, kinda left it on my branch 😅 
image

## Issues
Problem with use of credential provider for password auth, either switch to jwt or have some way to create session with prisma adapter

 * Temporary solution is to switch to jwt, ideally can store session token for password auth, but idt PrismaAdapter provides such functionality
 * IMPT: Have to add NEXT_AUTH_SECRET to the environment

<https://next-auth.js.org/providers/credentials>

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -34,3 +34,6 @@ NEXTAUTH_URL="http://localhost:3000"
 # Next Auth Discord Provider (lmao)
 DISCORD_CLIENT_ID=
 DISCORD_CLIENT_SECRET=
+
+# Next auth secret
+NEXT_AUTH_SECRET=1203jdifjmkldsmldas
```
Comment (Nafeij) [inline .env.example]: Is it possible to remove this secret? You can share it with us through private channels.

**Review by Nafeij — DISMISSED** [review]

File: .env.example (hunk 1 above)
Comment (Gabau) [reply .env.example]: It's just a random secret for jwt hashing, the text here is just an example, you can set it to what ever you like on your local dotenv

Comment (Gabau) [reply .env.example]: I'll switch it to something more meaningful

**Review by Nafeij — APPROVED** [review]
LGTM

**Pull request merged**